presto-main/src/test/java/com/facebook/presto/block/TestMapBlock.java

```diff
@@ -111,6 +113,15 @@ private void testWith(Map<String, Long>[] expectedValues)
         assertBlockFilteredPositions(expectedValues, block, () -> blockBuilder.newBlockBuilderLike(null), 0, 1, 3, 4, 7);
         assertBlockFilteredPositions(expectedValues, block, () -> blockBuilder.newBlockBuilderLike(null), 2, 3, 5, 6);
 
+        // Re-create a MapBlock without hashtables built to test getRegion.
+        Block blockForGetRegion = createBlockWithValuesFromKeyValueBlock(expectedValues);
+
+        int offset = (int) floor(blockForGetRegion.getPositionCount() / 2.0);
+        int length = (int) ceil(blockForGetRegion.getPositionCount() / 2.0);
+        Block halfBlock = blockForGetRegion.getRegion(offset, length);
+        Map<String, Long>[] expectedValuesInRegion = (Map<String, Long>[]) getObjectsInRegion(expectedValues, offset, length);
+        assertBlock(halfBlock, () -> blockBuilder.newBlockBuilderLike(null), expectedValuesInRegion);
```

Comment:
Make sure this tests failed without the fix you have . I am wondering if you want to test with the whole block after the `blockRegion` as well?